### PR TITLE
Fix issue 19043 -  Incorrect mangling for extern(C++) const template parameter on windows

### PR DESCRIPTION
**Pull request by thewilsonator**
I hope this works.

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]
Please squash commits when done.

**Comment by thewilsonator** [comment]
Will do, I wasn't expecting it to pass first time round (excluding my stupid mistakes).

**Comment by thewilsonator** [comment]
I've fixed this for msvc mangling, @WalterBright is it the same for DMC mangling?

**Comment by thewilsonator** [comment]
Why is DAutoTest failing and what can I do about it?

File: changelog/fix19043.dd [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Mangling of extern(C++) template on Windows now correctly mangles const non-pointer template parameters.
+It used to ignore the const and was missing the $$C escape prefix.
```
Comment (wilzbach) [inline changelog/fix19043.dd]: Read the error message ;-)

```
object.Exception@../tools/changed.d(216): Changelog entries should consist of one title line, a blank separator line, and a description.
```

**Review by Geod24 — APPROVED** [review]

**Pull request merged**